### DISCUSSION TITLE
IE 11 Issue + React Hook Form

### DISCUSSION THREAD

**Discussion by vbav** [react-hook-form/react-hook-form#3418] [Q&A] [answered]
I am new to React (created from CRA + Typescript Template) and integrated React Hook Form into our solution. For the most it went well untiI deployed this to IE11. At first my react app did not even load. 
**Adding below polyfills to my root index.tsx fixed the load issue:**
import 'react-app-polyfill/ie11';
import 'react-app-polyfill/stable';

Then I realized all the forms are not submitting. As suggested in FAQs, I did the following:
**Inside my components:**
import { useForm, Controller } from "react-hook-form/dist/index.ie11";

**Inside root Index.tsx, I imported the below polyfill:**
import 'es7-object-polyfill'; (after the above react polyfills)

Still no luck! Did I miss anything? I see no errors in console either. It is just that when I click on any button it does not do anything.

**Environment:**
"react": "^16.13.1",
"react-app-polyfill": "^1.0.6",
"typescript": "^4.0.5"
"react-hook-form": "^6.10.1"

IE Version: 11.1518.17763.0

**Comment by bluebill1049** [comment, marked as answer]
what's the error on the screen?

**Reply by vbav** [reply]
@bluebill1049 No error on console or screen. Clicking the Submit does not do anything. I tried putting a breakpoint inside the Submit callback function, it is not getting hit. Are the steps I followed correct?

Also, I am using the CRA Template, which I believe is internally using babel transpiler? Is that an issue? I noticed you guys are using the Typescript transpiler?

**Reply by bluebill1049** [reply]
if there is no error, then i am assuming it's working, can you try the following code: 
```
import React from "react";
import { useForm } from "react-hook-form";

export default function App() {
  const { register, handleSubmit, watch, errors } = useForm();
  const onSubmit = data => console.log(data);

  console.log(watch("example")); // watch input value by passing the name of it

  return (
    {/* "handleSubmit" will validate your inputs before invoking "onSubmit" */}
    <form onSubmit={handleSubmit(onSubmit)}>
    {/* register your input into the hook by invoking the "register" function */}
      <input name="example" defaultValue="test" ref={register} />
      
      {/* include validation with required or other standard HTML validation rules */}
      <input name="exampleRequired" ref={register({ required: true })} />
      {/* errors will return when field validation fails  */}
      {errors.exampleRequired && <span>This field is required</span>}
      
      <input type="submit" />
    </form>
  );
}
```

**Reply by vbav** [reply]
@bluebill1049 I see you are not using import from IE11 shim. Is that not required anymore?

import { useForm } from "react-hook-form";

**Reply by bluebill1049** [reply]
oh sorry just change that line to ie11 version

**Reply by vbav** [reply]
@bluebill1049 - I think I figured what the problem. It is mostly to do with my Button being outside my form (I was banging my head 6hrs trying all other crazy things). This is due to the structure of my Html fields. It seems to be a known issue in IE11. Interesting part the code you have given me seems to be working on IE11 without the shim. Is the shim specific to certain versions of IE11?

**Reply by bluebill1049** [reply]
not that i aware such shim.

**Reply by vbav** [reply]
Thanks for all your help and the great library! You will soon see a donation coming your way from me.

**Reply by bluebill1049** [reply]
@vbav that's too kind of you. Thank you very much.

**Reply by vbav** [reply]
@bluebill1049 - Just made the donation. Keep up the good work!

**Reply by bluebill1049** [reply]
Thank you very much again 🙏

**Comment by vbav** [comment]
Just posting it for others. The issue is my Button being outside the form and nothing to with this library. The form does not submit when the button is outside form when in IE 11 (this is a well known issue). It may sound stupid on my part but yet times we simply get carried away.
